application/src/main/java/org/togetherjava/tjbot/commands/UserInteractor.java

```diff
@@ -0,0 +1,84 @@
+package org.togetherjava.tjbot.commands;
+
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent;
+import net.dv8tion.jda.api.events.interaction.SelectionMenuEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.componentids.ComponentIdGenerator;
+
+import java.util.List;
+
+/**
+ * Represents a feature that can interact with users. The most important implementation is
+ * {@link SlashCommand}.
+ * <p>
+ * An interactor must have a unique name and can react to button clicks and selection menu actions.
+ */
+public interface UserInteractor extends Feature {
+
+    /**
+     * Gets the name of the interactor.
+     * <p>
+     * Requirements for this are documented in {@link CommandData#CommandData(String, String)}.
```

Comment:
Looks like broken copy paste

Reply:
which part is broken? looks good to me, no? I guess ur referring to the link? Well, unfortunately we dont have another place where we documented the name requirements :D

Reply:
UserInteractor doesn't sound like something which is Discord command specific
Those requirements only exist for commands that get sent to Discord (slash-commands, context-commands etc)

Reply:
yeah. but i think its better if we have the same requirements on all of them. after all, slash commands implement UserInteractor and use the same getName method.